### PR TITLE
Collection flyout removal

### PR DESCRIPTION
**Pull request by krupa**
From the add-on details page, we removed the section which allowed users to quickly add the current add-on to collections. This PR removes the related test and locators.

### CODE REVIEW

**Comment by stephendonner** [comment]
Thx @krupa r+

`(addon-tests) sdonners-MacBook-Pro:Addon-Tests sdonner$ py.test --driver=Firefox --base-url https://addons.allizom.org --servicesbaseurl https://services.addons-dev.allizom.org --variables /Users/sdonner/Desktop/AMO.json tests/desktop/test_details_page.py 
============================= test session starts ==============================
platform darwin -- Python 2.7.11, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /Users/sdonner, inifile: 
plugins: html-1.8.0, rerunfailures-0.5, selenium-1.2.1, variables-1.4, xdist-1.14
collected 28 items 

tests/desktop/test_details_page.py ..................s.........

==================== 27 passed, 1 skipped in 166.88 seconds ====================`

File: pages/desktop/details.py [hunk 1]
```diff
@@ -69,11 +69,6 @@ class Details(Base):
     _reviews_locator = (By.CSS_SELECTOR, "section#reviews div")
     _add_review_link_locator = (By.ID, "add-review")
 
-    _add_to_collection_locator = (By.CSS_SELECTOR, ".collection-add.widget.collection")
```
Comment (davehunt) [inline pages/desktop/details.py]: This is referenced in `click_add_to_collection_widget`, which should also be removed.

File: pages/desktop/details.py [hunk 2]
```diff
@@ -69,11 +69,6 @@ class Details(Base):
...
-    _add_to_collection_widget_button_locator = (By.CSS_SELECTOR, ".collection-add-login .register-button .button")
```
Comment (davehunt) [inline pages/desktop/details.py]: This is referenced in `collection_widget_button`, which should also be removed.

File: pages/desktop/details.py [hunk 3]
```diff
@@ -69,11 +69,6 @@ class Details(Base):
...
-    _add_to_collection_widget_login_link_locator = (By.CSS_SELECTOR, "div.collection-add-login p:nth-child(3) > a")
```
Comment (davehunt) [inline pages/desktop/details.py]: This is referenced in `collection_widget_login_link`, which should also be removed.

File: pages/desktop/details.py [hunk 4]
```diff
@@ -69,11 +69,6 @@ class Details(Base):
...
-    _add_to_favorites_widget_locator = (By.CSS_SELECTOR, 'div.widgets > a.favorite')
```
Comment (davehunt) [inline pages/desktop/details.py]: This is referenced in `click_add_to_favorites` and `is_addon_marked_as_favorite`. These are not used by the test `test_that_add_to_collection_flyout_for_anonymous_users` so removal of the locator will cause failures when calling these methods. Incidentally, these methods are referenced in `test_user_my_favorites_page` within `test_collections.py` which is currently failing due to element not found, so we may just need to remove that test too.

**Comment by davehunt** [comment]
This now has conflicts, probably due to #835

**Comment by davehunt** [comment]
This appears to have lost the latest changes. In an effort to get this landed and restore our automated suites, I have opened #838 with this rebased and squashed, whilst maintaining @krupa as the author.

**Pull request closed without merging**